module/data/actor/character.mjs

```diff
@@ -206,6 +206,55 @@ export default class CharacterData extends CreatureTemplate {
     }
     return data;
   }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Checks whether the item with the given relative UUID has been favorited
+   * @param {string} favoriteId  The relative UUID of the item to check.
+   * @returns {boolean}
+   */
+  hasFavorite(favoriteId) {
+    return !!this.favorites.find(f => f.id === favoriteId);
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Add a favorite item to this actor.
+   * If the given item is already favorite, this method has no effect.
+   * @param {object} favorite  The favorite to add.
+   * @returns {Promise<Actor5e>}
+   * @throws If the item intended to be favorited does not belong to this actor.
+   */
+  addFavorite(favorite) {
+    if ( this.hasFavorite(favorite.id) ) return Promise.resolve(this.parent);
+
+    if ( fromUuidSync(favorite.id, { relative: this.parent }) === null ) {
+      throw new Error(`The item with id ${favorite.id} is not owned by actor ${this.parent.id}`);
+    }
```

Comment:
This breaks dropping skills or tools to favourites.

Reply:
Ah, yeah that makes sense. I'll can come back and revisit that if you haven't already.

Reply:
I haven't had a chance, so feel free.